Sources/AsyncHTTPClient/HTTPClient+StructuredConcurrency.swift

```diff
@@ -0,0 +1,58 @@
+import NIO
+import Logging
+
+@available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
+extension HTTPClient {
+#if compiler(>=6.0)
+    /// Start & automatically shut down a new ``HTTPClient``.
+    ///
+    /// This method allows to start & automatically dispose of a ``HTTPClient`` following the principle of Structured Concurrency.
+    /// The ``HTTPClient`` is guaranteed to be shut down upon return, whether `body` throws or not.
+    ///
+    /// This may be particularly useful if you cannot use the shared singleton (``HTTPClient/shared``).
+    public static func withHTTPClient<R>(
+        eventLoopGroup: any EventLoopGroup = HTTPClient.defaultEventLoopGroup,
+        configuration: Configuration = Configuration(),
+        backgroundActivityLogger: Logger? = nil,
+        isolation: isolated (any Actor)? = #isolation,
+        _ body: sending (HTTPClient) async throws -> sending R
+    ) async throws -> sending R {
```

Comment:
Can we please use a more descriptive generic type name than `R`. It shows up in docs and autocomplete so it's worth making this better. Also the `body` must not be `sending`. `asyncDo` is not `sending` it anyways. This should run in the same task isolation as the caller of the `withHTTPClient` method

```suggestion
    public static func withHTTPClient<Result>(
        eventLoopGroup: any EventLoopGroup = HTTPClient.defaultEventLoopGroup,
        configuration: Configuration = Configuration(),
        backgroundActivityLogger: Logger? = nil,
        isolation: isolated (any Actor)? = #isolation,
        _ body: (HTTPClient) async throws -> sending Result
    ) async throws -> sending Result {
```